datadog/dogshell/service_level_objective.py

```diff
@@ -0,0 +1,418 @@
+        )
+        show_all_parser.add_argument(
+            "--query", help="string to filter SLOs by query (see UI or documentation)"
+        )
+        show_all_parser.add_argument(
+            "--slo_ids",
+            help="comma separated list indicating what SLO IDs to get at once",
+        )
+        show_all_parser.add_argument(
+            "--offset", help="offset of query pagination", default=0
+        )
+        show_all_parser.add_argument(
+            "--limit", help="limit of query pagination", default=100
+        )
+        show_all_parser.set_defaults(func=cls._show_all)
+
+        delete_parser = verb_parsers.add_parser("delete", help="Delete a SLO")
+        delete_parser.add_argument("slo_id", help="SLO to delete")
+        delete_parser.set_defaults(func=cls._delete)
+
+        delete_many_parser = verb_parsers.add_parser("delete_many", help="Delete a SLO")
+        delete_many_parser.add_argument(
+            "slo_ids", help="comma separated list of SLO IDs to delete"
+        )
+        delete_many_parser.set_defaults(func=cls._delete_many)
+
+        delete_timeframe_parser = verb_parsers.add_parser(
+            "delete_many", help="Delete a SLO timeframe"
+        )
+        delete_timeframe_parser.add_argument("slo_id", help="SLO ID to update")
+        delete_timeframe_parser.add_argument(
+            "timeframes",
+            help="CSV of timeframes to delete, e.g. 7d,30d,90d",
+            required=True,
+        )
+        delete_timeframe_parser.set_defaults(func=cls._delete_timeframe)
+
+    @classmethod
+    def _create(cls, args):
+        api._timeout = args.timeout
+        format = args.format
+
+        params = {"type": args.type, "name": args.name}
+
+        if args.tags:
+            tags = sorted(set([t.strip() for t in args.tags.split(",") if t.strip()]))
+            params["tags"] = tags
+
+        thresholds = []
+        for threshold_str in args.thresholds.split(","):
+            parts = threshold_str.split(":")
+            timeframe = parts[0]
+            target = parts[1]
+            warning = None
+            if len(parts) > 2:
+                warning = parts[2]
+            thresholds.append(
+                {"timeframe": timeframe, "target": target, "warning": warning}
+            )
+        params["thresholds"] = thresholds
+
+        if args.description:
+            params["description"] = args.description
+
+        if args.type == "metric":
+            params["query"] = {
+                "numerator": args.numerator,
+                "denominator": args.denominator,
+            }
+        elif args.monitor_search:
+            params["monitor_search"] = args.monitor_search
+        else:
+            params["monitor_ids"] = args.monitor_ids
+
+        if args.tags:
+            params["tags"] = args.tags.split(",")
+
+        res = api.ServiceLevelObjective.create(**params)
+        report_warnings(res)
+        report_errors(res)
```

Comment:
`report_warnings` and `report_errors` won't work here, since the key in the response is `error`, not `errors`. And I don't know if you even return `warnings`

Reply:
`warnings` is not currently returned, but keeping it consistent. `errors` is what is both reported and that function uses in the key

Reply:
right now it throws an exception when there are errors, so wondering if this should be an arg to the service implementation 🤔

Reply:
Actually yeah, you're right it raises an exception. I wonder how this even works in the other resources 🤔

Reply:
Oh I see, the `APIClient.submit` method swallows the exceptions.
But in your case, since you're raising when the `error` key in the response payload is set, directly in you resource, it will not be swallowed by the super class.
Are you saying that you can have responses which contain `errors` plural, as well as the `error` singular that you use to raise exception ?

Reply:
yes :) in some cases the API will return actual exceptions, in other cases we return an `errors` which can contain 0 or more errors depending on the endpoint. for bulk delete, some can succeed while others not, their reasons for not succeeding will be in `errors`

typically the responses look like `{"data": ..., "errors": ...}` even for HTTP409 for example

Reply:
added a workaround method